### DISCUSSION TITLE
Svelte: Typings on `subscription` method giving errors in TypeScript when using `TypedDocumentNode`

### DISCUSSION THREAD

**Discussion by V-ed** [urql-graphql/urql#1728] [Q&A] [answered]
moving issue https://github.com/FormidableLabs/urql/issues/1716 to this discussion / Q&A as it seems more appropriate here, especially if the error is from my usage of urql/svelte

As a quick summary of the issue above, I am getting type errors when using subscriptions in TypeScript when following the [official doc](https://formidable.com/open-source/urql/docs/advanced/subscriptions/#svelte) with `TypedDocumentNode`s variables, generated from https://www.graphql-code-generator.com/docs/plugins/typed-document-node

![image](https://user-images.githubusercontent.com/16652915/121937150-5f7f0080-cd18-11eb-9b2a-e1d3d3aaf2cc.png)

In the screenshot above, I'd expect `data` to be of type `NewMessagesSubscription` and my store not giving out errors...

Much more context is given in the issue above, for reference.

Any more help will be appreciated, thanks!

also cc @kitten (who followed up in the mentioned issue)

---

Changing the type in the `.d.ts` file from `Result` to `Data` as shown below :

```diff
- export declare function subscription<Data = any, Result = Data, Variables = object>(store: OperationStore<Result, Variables>, handler?: SubscriptionHandler<Data, Result>): OperationStore<Result, Variables>;
+ export declare function subscription<Data = any, Result = Data, Variables = object>(store: OperationStore<Data, Variables>, handler?: SubscriptionHandler<Data, Result>): OperationStore<Result, Variables>;
```

fixes my typing issue and correctly types the `data` parameter :
![image](https://user-images.githubusercontent.com/16652915/121942598-a8d24e80-cd1e-11eb-818f-56cb27042f29.png)

**Comment by V-ed** [comment]
My current solution is to redefine the type of the `subscription` method in my own project with this :

```ts
// /@types/urqlsvelte.d.ts

import { OperationStore, SubscriptionHandler } from '@urql/svelte';

declare module '@urql/svelte' {
	export declare function subscription<Data = any, Result = Data, Variables = object>(
		store: OperationStore<Data, Variables>,
		handler?: SubscriptionHandler<Data, Result>,
	): OperationStore<Result, Variables>;
}
```

Hopefully a cleaner solution rises up from my hair pulling research :)

**Comment by kitten** [comment, marked as answer]
oh so it's a typo all along! sorry, when I went over your original issue I assumed the generic was the same as in our other bindings and I must've misread it!

**Reply by V-ed** [reply]
aaah gotcha, maybe I didn't explain it correctly in my original issue, sorry about that! I think we can then follow up in the issue directly!

a little bit of back and forth but no worries :)
